codeops-scripts/apply-github-settings.ps1

```diff
@@ -0,0 +1,198 @@
+#Requires -Modules @{ ModuleName="powershell-yaml"; ModuleVersion="0.4.2" }
+
+#
+# This script is responsible for ensuring all GitHub repositories are
+# configured as per a policy of defined repository settings.
+#
+# For example, enforcing a branch protection.
+# 
+
+[CmdletBinding()]
+param (
+    [string] $ConfigDirectory,
+    [switch] $WhatIf
+)
+$ErrorActionPreference = 'Stop'
+
+$here = Split-Path -Parent $PSCommandPath
+
+# Install other module dependencies
+$requiredModules = @(
+    @{ Name = "Endjin.CodeOps"; Version = "0.2.2" }
+)
+$requiredModules | ForEach-Object {
+    if ( !(Get-Module -ListAvailable $_ | Where-Object { $_.Version -eq $_.Version }) ) {
+        Install-Module -Name $_.Name `
+                       -RequiredVersion $_.Version `
+                       -AllowPrerelease:($_.Version -match '-') `
+                       -Scope CurrentUser `
+                       -Repository PSGallery `
+                       -Force
+    }
+    Import-Module -Name $_.Name -RequiredVersion $_.Version
+}
+
+
+#
+# Helper functions
+#
+function _getAllOrgReposWithDefaults {
```

Comment:
Does this only deal with public repos?

Reply:
It will find all repos public or private, although the branch protection policy can only be applied to public repos.

What are your thoughts on whether to include the 'endjin' org?